### DISCUSSION TITLE
API (incident) Workflows?

### DISCUSSION THREAD

**Discussion by adamjack** [ulfsri/neris-framework#20] [Q&A]
As well as the "dispatch" question ( https://github.com/ulfsri/neris-framework/discussions/16 ) I have questions about the expected workflow with incidents and the API. Since I only see a "create incident" API some questions:

- When does the incident get filed? Some incident reports need information from the fire investigation, so the *final* incident (with all updates) is days/weeks/more delayed. Does Fire RMS not file until no more changes can be made? (There is a similar/extended timeline for casualties & their outcomes.) I see no "update incident".
- Further with no "lookup incident", and assuming "update incident" becomes available, how are changes made at the NERIS servers (e.g. from a mobile app or other source, e.g. CAD) included in the final incident w/o overwriting changes?

Thanks

Adam

**Comment by cweinschenk** [comment]
HI @adamjack,

Excellent questions.

- Part of the reason we have decoupled incident analysis from core incident fields is that there is now a natural split in the schemas and can be submitted independently. National engagement on the incident analysis (i.e., investigation) is open until 9/4/24. As far as the API is concerned, the `put` methods are currently being development. We chose to release the incident API with only the `post` method out so that additional development could begin while the remaining components of the API get finalized. Fire RMS can submit an incident once the department has closed it, and you will be able to update the record if for example a non-fatal casualty becomes a fatal casualty. Changes will be saved in the database.
- Similar to above, we have be working on what are the primary data fields that should be queryable and will release those updates to the API as soon as they are ready.

-Craig
